market/src/main/java/com/banica/market/configurations/MarketConfigurationImpl.java

```
@@ -178,27 +178,69 @@ public void updateGoodSpecification(String name, Long quantityLow, Long quantity
+    public void updateSnapshotFrequency(int seconds) {
+
+        if (seconds < 0) {
+            logger.warn("Snapshot frequency seconds must be a positive number");
+            throw new IllegalArgumentException();
+        }
+        Properties props = new Properties();
+
+        modifyPropertySource(Collections.singletonMap("snapshot-frequency-seconds", String.valueOf(seconds)),
+                props, props::setProperty);
+
+        snapFrequency = seconds;
+    }
+
     public void setGoodSpecificationMap(Map<String, GoodSpecification> goodSpecificationMap) {
         this.goodSpecificationMap = goodSpecificationMap;
     }
 
+    public void setSnapFrequency(int snapFrequency) {
+        this.snapFrequency = snapFrequency;
+    }
+
+    public int getSnapFrequency() {
+        return snapFrequency;
+    }
+
     private void validate(GoodSpecification specifications) {
         if (!validator.test(specifications)) {
             logger.warn("GoodSpecification {} doesn't meet validation constraints.", specifications);
             throw new IllegalArgumentException();
         }
     }
 
-    private void modifyPropertySource(GoodSpecification goodSpecification,Properties properties,
+    private void modifyPropertySource(Map<String, String> propertyMap, Properties properties,
                             BiConsumer<String, String> modifyingFunction) {
-        try {
-            properties.load(new FileInputStream(file));
-            goodSpecification.createPropertyMap("good-specification-map.")
-                    .forEach(modifyingFunction);
-            properties.store(new FileOutputStream(file), "");
+
+        if (!file.exists()) {
+            try {
+                Files.createDirectories(Paths.get(file.getParent()));
+                Files.createFile(Paths.get(file.getPath()));
+            } catch (IOException e) {
+                e.printStackTrace();
+                return;
```

On a second thought, it might be even better to throw an exception back to JMX and reject the operation.